### DISCUSSION TITLE
How to run pipeline on acquisition/ subject class

### DISCUSSION THREAD

**Discussion by VeroniqueChouinard** [LabeoTech/Umit#74] [General]
Hello Bruno, how are you?

I saw that the toolbox integrates the possibility on running the pipeline on the acquisition class and subject as well. I would like to try running the pipeline on the acquisition class instead of modality like I usually do but i don't know how to go about it. 
To be precise, my idea is to produce an average map of multiple acquisitions merged together. I do tactile stimulation of the forelimb, hindlimb and whiskers and I would like to be able to produce images of these three or four acquisitions together. 

Is there a documentation or examples available to do that?

**Comment by BrunoOFSouza** [comment]
Hi Veronique,

If I understood correctly, you want to create an average of the activation maps from different acquisitions of a given mouse. There is a way to do it by using a combination of existing functions and tools inside the **umIToolbox** app. 

*The following workflow assumes that the data was already preprocessed and aligned

Here is a workflow that could work for you:

1) Create the amplitude map for each acquisition using the [**calculate_response_amplitude**](https://labeotech.github.io/Umit/documentation/userDocs/fcns/calculate_response_amplitude.html) function.
2) Save the amplitude maps in *.mat* file format using the function [**transformDat2Mat**](https://labeotech.github.io/Umit/documentation/userDocs/fcns/transformDat2Mat.html). This will allow you to visualize the maps in the *Analysis* tab of the **umIToolbox** app.
3) In the *Analysis* tab, select all recordings that you want to average, select the files containing the maps and click on the *Select* button ([doc](https://labeotech.github.io/Umit/documentation/userDocs/apps/umIToolbox-Analysis.html#how-to-select-data-for-analysis)).
4) Now, use the utility [*Avg. Acquisitions*](https://labeotech.github.io/Umit/documentation/userDocs/apps/umIToolbox-Analysis-AvgAcqs.html) to average all maps.
5) Finally, launch the [*Plot Average Maps*](https://labeotech.github.io/Umit/documentation/userDocs/tools/PlotAvgMaps_tool.html) to visualize.

Let us know if this doesn't work.

**Comment by VeroniqueChouinard** [comment]
Thank you, it works perfectly! However, I noticed that in the analysis tab when I select another file by mistake (which I do lot more than I should), Matlab sends the following error : 

Error using umIToolbox/SelectFileListBox_analysisValueChanged
'obsID' does not exist in 'F:\BlindMouse_Project2024\Data\Processed\4ADG\iso-stim-lFL-bs-1-3\CtxImg\data_splitByEvent.mat'.

The toolbox then remains in 'please wait' mode until I close it. It's not a huge problem for me since I just have to close and reopen the GUI, but i thought I should still report the bug.

**Reply by BrunoOFSouza** [reply]
Looks like a Bug. Which version of the toolbox are you using?

Thanks!

**Reply by VeroniqueChouinard** [reply]
Bipolar- v2.0.0
